src/extension.ts

```diff
@@ -1,18 +1,57 @@
+import { FileDecorationProvider } from "./FileDecorationProvider";
 import { SidebarProvider } from "./SidebarProvider";
+import { filesManager, stateManager } from "./context";
+import { Map } from "./types/extensionTypes";
+import {
+  USER_READY,
+  FILE_CLICK,
+  CURSOR_MOVE,
+  USER_LEAVE,
+  CURRENT_POSITION,
+} from "./utils/constants";
+// Initialize env vars
+import "dotenv/config";
+import { io } from "socket.io-client";
 import * as vscode from "vscode";
 
-// This method is called when your extension is activated
-// Your extension is activated the very first time the command is executed
+const path = require("path");
+require("dotenv").config({ path: path.resolve(__dirname, "..", ".env") });
+
+// This method is called when your extension is activated.
+// Currently this is activated as soon as user open VSCode
 export function activate(context: vscode.ExtensionContext) {
-  // Initialize the Sidebar
-  const sidebarProvider = new SidebarProvider(context.extensionUri);
+  // Autosave
+  const config = vscode.workspace.getConfiguration("files");
+  config.update("autoSave", "afterDelay", true);
+  config.update("autoSaveDelay", 100, true);
+
+  const sessionId = process.env.SESSION_ID || "";
+  const socket = io(
+    "https://socket-server-dot-live-sessions-staging.uc.r.appspot.com"
```

Comment:
What do you think about making this an env var? Since this will be user-facing I plan on having separate deployments for staging and prod